Hotfix 1.0.65 copy & legal changes -> Develop

JacobJaffe opened the pull request:
#### Description:

Adds the copy changes from master (1.0.65) branch to develop so we can unblock BT team. 
Changes:
- legal screen 
- HA selection screen 
- on-boarding flow (removes `Onboarding4`)
- `.en` and `.ht` string files

> NOTE: this defaults strings to `1.0.65`'s version, except for the string `label.covid_safe_paths` on the legal screen, which was added in develop branch. 

I also did some cleanup on the navigation `entry.js` & onboarding screens while touching those.

On the legal screen, I added a safe area context to the bottom button (seems like an important button to ensure is clickable across devices).

@efalkner lmk if we need a version bump on this

#### Linked issues:
https://pathcheck.atlassian.net/browse/SAF-406

#### Screenshots:

<img width="570" alt="image" src="https://user-images.githubusercontent.com/25315679/82387952-51bc5600-9a06-11ea-852c-53a8a32cadeb.png">
<img width="570" alt="image" src="https://user-images.githubusercontent.com/25315679/82387931-4bc67500-9a06-11ea-81e4-02304beab86e.png">


#### How to test:

We need to ensure that the updated strings are good to go -- these default to the hotfix version, so if develop has updated any, those will be lost. 

Generally, manual test over the onboarding flow, legal page, and HA page. 

Ensure string populate in Haitian Creole also.

JacobJaffe committed:
Copy & onboarding screen namings

JacobJaffe committed:
About screen

JacobJaffe committed:
choose provider screen & relavent strings

JacobJaffe committed:
legal screen

JacobJaffe committed:
snapshots

JacobJaffe commented on gps/app/Entry.js:
```diff
@@ -55,94 +55,35 @@ class Entry extends Component {
             cardStyle: {
               backgroundColor: 'transparent', // prevent white flash on Android
             },
+            headerShown: false,
```
Popped this screen option to the highest level to make this file easier to read

JacobJaffe commented on gps/app/Entry.js:
```diff
@@ -31,15 +30,16 @@ class Entry extends Component {
     super(props);
     this.state = {
-      initialRouteName: '',
+      onboardingComplete: null,
```
this state piece has been used as a flag to determine this, so made the naming explicit

JacobJaffe committed:
string updates

JacobJaffe commented on gps/app/locales/en.json:
```diff
@@ -5,22 +5,18 @@
     "operating_system_abbr": "OS:",
```
this file has just been copied from `1.0.65`

tstirrat replied:
This is probably not quite enough. keys likely changed a lot.

Do a `yarn i18n:extract` and you'll see it add/remove keys based on all keys found in the js views.

Anything that is added/removed here will be added/removed from the translation mgmt system Lokalise. So we should be extra careful here before submitting.

JacobJaffe commented on gps/app/locales/ht.json:
```diff
@@ -1,82 +1,99 @@
 {
```
this file has just been copied from `1.0.65`

tstirrat replied:
Same here, once you do a `yarn i18n:extract` (which only updates en.json) you'll need to compare keys with en.json.

JacobJaffe replied:
Ah very cool. Makes sense, doing that now.

JacobJaffe commented on gps/app/views/About.js:
```diff
@@ -10,10 +10,10 @@ import {
 } from 'react-native';
```
this file has just been copied from `1.0.65`

JacobJaffe commented on gps/app/views/Licenses.js:
```diff
@@ -1,25 +1,24 @@
+/* eslint-disable react-native/no-raw-text */
 import React, { useEffect } from 'react';
```
this file has been copied from `1.0.65`.
I also added a safe area view for the bottom button

JacobJaffe commented on gps/app/views/onboarding/Onboarding1.js:
```diff
@@ -26,7 +26,7 @@ import { sharedStyles } from './styles';
 
 const width = Dimensions.get('window').width;
 
-class Onboarding extends Component {
```
Made these match their file names

JacobJaffe commented on gps/app/views/onboarding/Onboarding1.js:
```diff
@@ -99,7 +99,7 @@ class Onboarding extends Component {
               <View style={sharedStyles.footerContainer}>
                 <EulaModal
                   continueFunction={() =>
-                    this.props.navigation.replace('Onboarding2')
```
Best practices on preserving navigation stack. Lmk if this was intentional.

tstirrat replied:
I think it was intentional because it's an overlay on the first screen. If the back button presses dont feel wonky, its ok to change.

JacobJaffe replied:
👍 that's a good reason. Will keep

JacobJaffe committed:
snapshot updates

tstirrat commented on ios/Podfile.lock:
```diff
@@ -510,4 +510,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: ea69c9e97838a7efc940277c11da296608ea8ae1
 
-COCOAPODS: 1.9.1
+COCOAPODS: 1.8.4
```
update your cocoapods with `cd ios && bundle`

tstirrat commented on gps/app/views/onboarding/Onboarding4.js:
```diff
@@ -17,7 +17,7 @@ import { sharedStyles } from './styles';
 const width = Dimensions.get('window').width;
 
-const Onboarding = props => {
+const Onboarding4 = ({ navigation }) => {
```
if 4 never shows. let's remove it?

tstirrat reviewed (changes requested):
Mostly concerned about the key mapping for en.json

JacobJaffe committed:
navigate -> replace

JacobJaffe committed:
locale  updates

JacobJaffe committed:
eslint action workflow update

JacobJaffe committed:
remove empty key from .ht file, fix eslint yaml

tstirrat commented on .github/workflows/lint.yml:
```diff
@@ -24,7 +24,7 @@ jobs:
 
       # v1 is broken, the given sha was the latest master (but we dont want to track master)
-      - uses: tinovyatkin/action-eslint@8b32772f075413f9192f43641e03d9bd0abae621
+      - uses: hallee/eslint-action@master
```
do these in other PRs ideally. its unclear why this is needed.

tstirrat replied:
I already tried `hallee/*` and it failed miserably.

JacobJaffe replied:
Sorry what's the requested change here?
Don't change the lint.yml so other branches use prev eslint?

tstirrat commented on gps/app/views/ChooseProvider.js:
```diff
@@ -408,11 +409,23 @@ const styles = StyleSheet.create({
     alignSelf: 'center',
     backgroundColor: Colors.WHITE,
   },
+  startLoggingButtonTouchable: {
```
If you can make the check name "eslint" and not "ESlint check" it will show under the right place in actions: 

At the moment it shows in i18n: https://github.com/Path-Check/covid-safe-paths/pull/872/checks?check_run_id=691207449

JacobJaffe replied:
Are there two eslint checks? I see that one and also the "lint" action

tstirrat replied:
To clarify, There is only one eslint check, but the new action we used sets the annotations source to a different job name and so it doesnt match up..  so it looks like there are two checks.

The prior eslint checker maps the it's same job name (via with: config), so it lines up better.

Problem is it doesn't tell you in the job log what failed. ... grr.

JacobJaffe committed:
legal copy Path Check -> PathCheck

JacobJaffe committed:
snapshot test

JacobJaffe committed:
undo lint changes

JacobJaffe committed:
merge conflicts

JacobJaffe committed:
eslint for raw text fix & fix check image

JacobJaffe committed:
snapshot update

JacobJaffe committed:
remove podfile changes

JacobJaffe commented:
Reconciled latest development changes. Added a bug fix that was introduced in the icon switching that caused an icon not to appear, and fixed an eslint rule to allow the `Typography` component to use raw-text, so we don't disable the check on files.

tstirrat commented on .eslintrc.js:
```diff
@@ -67,6 +67,7 @@ module.exports = {
     'react/react-in-jsx-scope': 2, // Prevent missing React when using JSX
     'react/self-closing-comp': 2, // Prevent extra closing tags for components without children,
     'react/prefer-stateless-function': 2, // Use functional components vs classes
+    'react-native/no-raw-text': [2, { skip: 'Typography' }],
```
This could be a problem, it assumes `<Typography>` translates things, which it doesn't.

Prefer the inline overrides so that the developer can put a reason why such a string it is not translated.

In the case of feature flags toggle screen, we're ok not putting this burden on translators for something not seen outside in the real world.

tstirrat commented on app/views/ChooseProvider.js:
```diff
@@ -17,22 +17,19 @@ import {
 
-import { Icons } from '../assets';
-import { Button } from '../components/Button';
+import closeIcon from '../assets/images/closeIcon.png';
```
This is reverting the icon constants work that was done recently.

tstirrat reviewed (dismissed):
Generally approve, but revert the eslint override.

JacobJaffe committed:
eslint revert on raw text

JacobJaffe committed:
reset eslint to develop

JacobJaffe commented on app/assets/images/index.js:
```diff
@@ -1,7 +1,7 @@
-import BoxCheckedIcon from './boxCheckedIcon.png';
-import boxUncheckedIcon from './boxUncheckedIcon.png';
+import BoxChecked from './boxCheckedIcon.png';
+import BoxUnchecked from './boxUncheckedIcon.png';
```
the capitalization was causing a bug in a consumer

tstirrat commented:
The remaining lint issues are: 

```
yarn run v1.22.4
$ eslint ./

/Users/tstirrat/dev/safe-paths/app/views/ChooseProvider.js
  417:32  warning  Color literal: { backgroundColor: '#665eff' }  react-native/no-color-literals
  425:27  warning  Color literal: { color: '#ffffff' }            react-native/no-color-literals
  429:3   warning  Unused style detected: styles.headerTitle      react-native/no-unused-styles
  453:20  warning  Color literal: { borderColor: '#999999' }      react-native/no-color-literals

✖ 4 problems (0 errors, 4 warnings)
```

My pr should make this clearer #886

JacobJaffe commented:
Now outdated from merges in throughout the week separately addressing this.
